patches/server/0141-Basic-PlayerProfile-API.patch

```diff
@@ -103,14 +103,14 @@ index 0000000000000000000000000000000000000000..e513ce5bcb5070b8a57737228904381f
 +    @Nullable
 +    @Override
 +    public UUID getId() {
-+        return profile.getId();
++        return profile.getId() != Util.NIL_UUID ? profile.getId() : null;
 +    }
 +
 +    @Override
 +    @Deprecated(forRemoval = true)
 +    public UUID setId(@Nullable UUID uuid) {
 +        GameProfile prev = this.profile;
-+        this.profile = new GameProfile(uuid, prev.getName());
++        this.profile = new GameProfile(uuid != null ? uuid : Util.NIL_UUID, prev.getName());
 +        copyProfileProperties(prev, this.profile);
 +        return prev.getId();
```

Comment:
This maybe needs to be updated to avoid a contract change from nullable to notnull?